packages/create-react-app-demo/src/ai.tsx

```diff
@@ -0,0 +1,175 @@
+import * as LLMx from '@fixieai/ai-jsx';
+import React from './react.ts';
+import { useState } from 'react';
+import { Suspense, ReactNode, useRef, useEffect } from 'react';
+import _ from 'lodash';
+import {
+  Recipe,
+  RecipeIngredientList,
+  RecipeInstructionList,
+  RecipeIngredientListItem,
+  RecipeInstructionListItem,
+  RecipeTitle,
+  SelectIngredientsButton,
+} from './recipe/page.tsx';
+
+export function useAI(children: LLMx.Node, dependencies: unknown[], when: boolean = true) {
+  const isInProgressRef = useRef(false);
+  const mostRecentlyRenderedChildren = useRef(children);
+  // If `children` changes, but a previous call is still in progress, will we properly start a new one?
+  const [result, setResult] = React.useState('');
+  const [isDone, setIsDone] = React.useState(false);
+
+  useEffect(() => {
```

Comment:
IIUC this is client-side -- because of the API key problem I'm not sure fully client-side will ever be a first class scenario, right?